### PR TITLE
Parallelize termset_pairwise_similarity() calls

### PR DESCRIPTION
**Pull request by justaddcoffee**
Hacking with @caufieldjh et al to parallize calls to `termset_pairwise_similarity()` in `associations_search()`

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -185,11 +185,6 @@ impl RustSemsimian {
         minimum_jaccard_threshold: &Option<f64>,
         minimum_resnik_threshold: &Option<f64>,
     ) -> SimilarityMap {
-        let pb = generate_progress_bar_of_length_and_message(
```
Comment (hrshdhgd) [inline src/lib.rs:188]: Instead of deleting, should we just comment out this code? That way if we need it while debugging, it'll be helpful!

**Review by hrshdhgd — COMMENTED** [review]
Did we check benchmarking on this code?

**Comment by justaddcoffee** [comment]
> Did we check benchmarking on this code?

Yep, it shaves a little time off:
```
search_bench_group/search
                        time:   [24.153 s 24.293 s 24.419 s]
                        change: [-11.239% -9.5374% -8.2260%] (p = 0.00 < 0.05)
                        Performance has improved.
```

**Comment by justaddcoffee** [comment]
> Instead of deleting, should we just comment out this code? That way if we need it while debugging, it'll be helpful!

Yeah, or actually we could add it back - I'd forgotten I got rid of it. This PR is really just an attempt to parallelize the `parallelize_termset_pairwise_similarity()` calls

**Comment by justaddcoffee** [comment]
> Instead of deleting, should we just comment out this code? That way if we need it while debugging, it'll be helpful!

just added this back (commented)

**Review by hrshdhgd — APPROVED** [review]

**Pull request merged**